pyk/src/pyk/ktool/kprove.py

```diff
@@ -110,23 +107,22 @@ def __init__(self, definition_dir: Path, main_file: Optional[Path] = None, use_d
 
     def prove(
         self,
-        spec_file,
-        spec_module_name=None,
-        args=[],
-        haskell_args=[],
-        haskell_log_entries=[],
-        log_axioms_file=None,
-        allow_zero_step=False,
-        dry_run=False,
-        rule_profile=None,
+        spec_file: Path,
+        spec_module_name: Optional[str] = None,
+        args: Iterable[str] = (),
+        haskell_args: Iterable[str] = (),
+        haskell_log_entries: Iterable[str] = (),
+        log_axioms_file: Optional[Path] = None,
+        allow_zero_step: bool = False,
+        dry_run: bool = False,
     ):
         log_file = spec_file.with_suffix('.debug-log') if log_axioms_file is None else log_axioms_file
         if log_file.exists():
             log_file.unlink()
-        haskell_log_entries += ['DebugTransition']
-        haskell_log_entries += ['DebugAttemptedRewriteRules'] if rule_profile else []
-        haskell_log_entries = unique(haskell_log_entries)
-        haskell_log_args = ['--log', str(log_file), '--log-format', 'oneline', '--log-entries', ','.join(haskell_log_entries)]
+        _haskell_log_entries = list(haskell_log_entries)
+        _haskell_log_entries += ['DebugTransition']
+        _haskell_log_entries = list(unique(_haskell_log_entries))
+        _haskell_log_args = ['--log', str(log_file), '--log-format', 'oneline', '--log-entries', ','.join(_haskell_log_entries)]
```

Comment:
Whats the reason for adding the underscore here?

Reply:
typing

Reply:
The type of `haskell_log_entries` is an immutable `Iterable` as parameters, can't add to it I don't think. So just make a new mutable list `_haskell_log_entries` and use that.

Reply:
E.g. This types fine https://github.com/runtimeverification/k/blob/kit-upstream-2/pyk/src/pyk/ktool/kompile.py#L39

Reply:
Ah you're right. I adjusted it.